### PR TITLE
[WIP] updating faraday and middleware versions

### PR DESCRIPTION
**Pull request by d-m-u**

### CODE REVIEW

**Comment by d-m-u** [comment]
oh I don't know
@bdunne could you please review? or anyone?

**Comment by agrare** [comment]
@d-m-u specs are failing

**Comment by d-m-u** [comment]
@agrare, fine, want to take another look?

**Comment by abellotti** [comment]
strange that we're pulling in an RC version for the middleware, is that the intent ? (i.e. 0.14.0 wouldn't help us here)

**Comment by agrare** [comment]
API client isn't my thing :) I'll let @bdunne and @abellotti take a look

**Comment by d-m-u** [comment]
@abellotti want me to wait for one that isn't rc? 0.14.0 will not work for this, no. it's faraday is capped at < 1.0.0 and we need at minimum the 1.0 release.

**Comment by abellotti** [comment]
Can we wait ?  that'd be preferred, i.e. having all the kinks out before we pull it into production here.
/cc @bdunne thoughts ?

**Comment by abellotti** [comment]
btw, I have the latest local gems installed and client seems to work fine with 0.14.0 of the middleware.

faraday (1.0.0, 0.17.3, 0.17.0, 0.9.2)
faraday_middleware (0.14.0, 0.13.1, 0.10.1)

ManageIQ::API::Client.new.vms.where(:name => "aab*").collect(&:name)

what was it that 1.0.0.rc1 fixes for us ?

**Comment by d-m-u** [comment]
We're been putting off updating Faraday for _years_, and it's time to update to 1.0.0.

**Comment by abellotti** [comment]
you're right.

image

while it runs fine with it locally, the faraday_middleware gem dependency fails.

**Comment by abellotti** [comment]
looks like it's very close to becoming official, https://github.com/lostisland/faraday_middleware/issues/200    folks are kicking the tire.  I'd wait before we pull it in to production.

File: manageiq-api-client.gemspec [hunk 1]
```diff
@@ -28,8 +28,8 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "rspec", "~> 3.0"
 
   spec.add_dependency "activesupport", ">= 5.0", "< 5.3"
-  spec.add_dependency "faraday", "~> 0.9"
-  spec.add_dependency "faraday_middleware", "~> 0.10.0"
+  spec.add_dependency "faraday", "~> 1.0"
+  spec.add_dependency "faraday_middleware", ">= 1.0.0.rc1", "< 2.0.0"
```
Comment (bdunne) [inline manageiq-api-client.gemspec]: I'd prefer not to ship with an RC

**Comment by bdunne** [comment]
Is there a source of urgency behind this PR? Like a security issue or something?

**Comment by d-m-u** [comment]
No, it's fine. It's marked as WIP. I'm just impatient.

**Comment by d-m-u** [comment]
Closing because I expect that with the license change to vcr we can't use Faraday 1.0.0 anyway. Per conversations in gitter recently and with Jason the new license will prohibit RH from using the latest VCR and so this update isn't going to happen.

**Comment by iMacTia** [comment]
@d-m-u FaradayMiddleware 1.0 has now been released: https://rubygems.org/gems/faraday_middleware/versions/1.0.0

**Pull request closed without merging**